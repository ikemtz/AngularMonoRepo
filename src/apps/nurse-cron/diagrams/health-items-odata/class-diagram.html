<!DOCTYPE html>
<html lang="en">
  <head>
    <title>Mermaid Class Diagram - by openapi-mermaid</title>
    <meta content="text/html; charset=utf-8" />
    <script src="https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js"></script>
    <script>mermaid.initialize({startOnLoad:true});</script>
  </head>
  <body>
    <div class="mermaid">
classDiagram


  class HealthItem{
    +uuid: id
    +string: name
    +bool: isEnabled
    +string: createdBy
    +string: updatedBy
    +date: createdOnUtc
    +date: updatedOnUtc
  }
    </div>
    <div style="font-family: Arial, Helvetica, sans-serif; font-size: small;">
      This file was generated by the <a href="https://www.npmjs.com/package/openapi-mermaid">openapi-mermaid</a> tool.
    </div>
  </body>

  </body>
</html>
