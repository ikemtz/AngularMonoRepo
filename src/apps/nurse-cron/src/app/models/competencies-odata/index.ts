/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.form';
export * from './competency.model';
export * from './competency.properties';
export * from './competency.test-obj-fac';
