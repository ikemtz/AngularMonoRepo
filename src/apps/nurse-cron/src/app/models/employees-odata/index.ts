/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './employee-certification.form-group-fac';
export * from './employee-certification.form';
export * from './employee-certification.model';
export * from './employee-certification.properties';
export * from './employee-certification.test-obj-fac';
export * from './employee-competency.form-group-fac';
export * from './employee-competency.form';
export * from './employee-competency.model';
export * from './employee-competency.properties';
export * from './employee-competency.test-obj-fac';
export * from './employee-health-item.form-group-fac';
export * from './employee-health-item.form';
export * from './employee-health-item.model';
export * from './employee-health-item.properties';
export * from './employee-health-item.test-obj-fac';
export * from './employee.form-group-fac';
export * from './employee.form';
export * from './employee.model';
export * from './employee.properties';
export * from './employee.test-obj-fac';
