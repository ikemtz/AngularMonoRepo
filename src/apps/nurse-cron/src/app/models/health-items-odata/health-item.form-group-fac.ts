/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function HealthItemFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    name: new FormControl(''),
    isEnabled: new FormControl(''),
  });
}
