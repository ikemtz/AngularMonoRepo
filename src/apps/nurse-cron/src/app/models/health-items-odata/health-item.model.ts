/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IHealthItem {
  id?: string;
  name?: string;
  isEnabled?: boolean;
}
