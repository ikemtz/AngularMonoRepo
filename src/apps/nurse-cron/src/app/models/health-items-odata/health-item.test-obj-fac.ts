/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #test-object-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */

import { HealthItemProperties } from './health-item.properties';

export function createTestHealthItem() {
    return { 
      [HealthItemProperties.ID]: 'ID',
      [HealthItemProperties.NAME]: 'NAME',
      [HealthItemProperties.IS_ENABLED]: false, 
    };
}
