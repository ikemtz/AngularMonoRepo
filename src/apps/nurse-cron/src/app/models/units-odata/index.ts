/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './building.form-group-fac';
export * from './building.form';
export * from './building.model';
export * from './building.properties';
export * from './building.test-obj-fac';
export * from './unit.form-group-fac';
export * from './unit.form';
export * from './unit.model';
export * from './unit.properties';
export * from './unit.test-obj-fac';
