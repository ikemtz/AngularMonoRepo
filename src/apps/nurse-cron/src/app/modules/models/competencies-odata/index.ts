/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.model';
export * from './competency.properties';
