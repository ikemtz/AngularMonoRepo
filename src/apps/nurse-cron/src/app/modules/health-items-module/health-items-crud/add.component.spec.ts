import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';

import { HealthItemAddComponent } from './add.component';
import { HealthItemCrudFacade } from './crud.facade';
import { HealthItemProperties, IHealthItem } from '../../../models/health-items-odata';

describe('HealthItemAddComponent', () => {
  let component: HealthItemAddComponent;
  let fixture: ComponentFixture<HealthItemAddComponent>;
  let facade: HealthItemCrudFacade;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [HealthItemAddComponent],
      imports: [ReactiveFormsModule, NoopAnimationsModule],
      providers: [{ provide: HealthItemCrudFacade, useValue: createDataEntryMockFacade() }],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(HealthItemAddComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(HealthItemCrudFacade);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should save', () => {
    component.initForm();
    component.addEditForm.patchValue({
      [HealthItemProperties.ID]: 'ID',
      [HealthItemProperties.NAME]: 'NAME',
      [HealthItemProperties.IS_ENABLED]: true,
    });

    let item: IHealthItem;
    facade.saveNewEntity = jest.fn(x => (item = x));
    facade.updateExistingEntity = jest.fn();
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(1);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);

    expect(item).toMatchSnapshot();

  });

  it('should not save', () => {
    component.addEditForm = { valid: false } as never;
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);
  });

  it('should cancel', () => {
    facade.clearCurrentEntity = jest.fn();
    component.cancel();
    expect(facade.clearCurrentEntity).toBeCalledTimes(1);
  });
});
