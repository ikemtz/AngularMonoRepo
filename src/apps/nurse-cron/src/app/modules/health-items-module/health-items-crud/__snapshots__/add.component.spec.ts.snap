// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthItemAddComponent should save 1`] = `
Object {
  "id": undefined,
  "isEnabled": true,
  "name": "NAME",
}
`;
